src/xb-silo.c

```diff
@@ -58,13 +58,13 @@ typedef struct {
 G_DEFINE_TYPE_WITH_PRIVATE (XbSilo, xb_silo, G_TYPE_OBJECT)
 #define GET_PRIVATE(o) (xb_silo_get_instance_private (o))
 
-enum {
-	PROP_0,
-	PROP_GUID,
+typedef enum {
+	PROP_GUID = 1,
 	PROP_VALID,
 	PROP_ENABLE_NODE_CACHE,
-	PROP_LAST
-};
+} XbSiloProperty;
+
+static GParamSpec *obj_props[PROP_ENABLE_NODE_CACHE + 1] = { NULL, };
```

Comment:
not PROP_LAST? That seemed a nice thing.

Reply:
No, because `PROP_LAST` is not really a member of the enum, it’s metadata about the enum. Unfortunately there’s no way in C to do something like `number_of_elements_in (XbSiloProperty)`. `PROP_LAST` is a hack to achieve that, but if used with `-Wswitch-enum` it means that every `switch` statement which handles values of that enum type will now have to have a `case PROP_LAST: g_assert_not_reached()` case, which is painful.

Using `PROP_ENABLE_NODE_CACHE + 1` in the array definition here is the least-worst solution I know. The compiler should catch if we fail to update the definition in future, since we’d be statically assigning to an element off the end of the array in `class_init` if so.